src/components/App/Shows/Show/index.js

```diff
@@ -0,0 +1,40 @@
+import React, {Component} from 'react';
+
+import styles from './Show.module.scss';
+
+class Show extends Component {
+  state = {
+    name: '',
+    imgUrl: '',
+  };
+
+  static getDerivedStateFromProps(nextProps, prevState) {
```

Comment:
I called this method because `componentWillReceiveProps` is going to be deprecated in future versions of React. The thing here is that I use the props but I need to modify them. So first, I need to turn props into the state, I cannot modify props because they are immutable. Second, the props will change after the async call to the API. Plus, because I noticed that I will need `https` protocol for future deployment do GH pages, I needed to manually change the `http` to `https` to avoid [mixed content issues](https://developers.google.com/web/fundamentals/security/prevent-mixed-content/fixing-mixed-content). That's why I decided to give this new lifecycle method a try. However, it's the first time I've done it and I am not sure if it's correct. Please advice on that. Please have a look here: https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops Am I drawing proper conclusions?

Reply:
It's ok for now. I believe that this method will dissapear after introduced `react-redux`